lib/cli/interactive-setup/service.js

```diff
@@ -55,27 +58,69 @@ const projectNameInput = async (workingDir) =>
     })
   ).projectName.trim();
 
+const resolveProjectNameInput = async (options, workingDir) => {
+  if (options.name) {
+    if (!isValidServiceName(options.name)) {
+      throw new ServerlessError(INVALID_PROJECT_NAME_MESSAGE, 'INVALID_PROJECT_NAME');
+    }
+
+    let alreadyTaken = false;
+    try {
+      await fs.promises.access(join(workingDir, options.name));
+      alreadyTaken = true;
+    } catch {
+      // Pass
+    }
+
+    if (alreadyTaken) {
+      throw new ServerlessError(
+        `Path ${options.name} is already taken`,
+        'TARGET_FOLDER_ALREADY_EXISTS'
+      );
+    }
+
+    return options.name;
+  }
+
+  return projectNameInput(workingDir);
+};
+
 module.exports = {
   isApplicable({ serviceDir }) {
     return !serviceDir;
   },
   async run(context) {
     const workingDir = context.cwd || process.cwd();
+    const options = context.options || {};
     const isConfirmed = await confirm('No project detected. Do you want to create a new one?', {
```

Comment:
I think this question should be skipped if any of those option is passed (it's a clear indication, that intention is to create a new project)